File: src/cpp/driver.cpp
#include <iostream>
using namespace std;

void robogo_sort(int*);
void bogo_sort(int*);

const int N = 8;

int main() {
    int list[] {1, 4, 2, 3, 5, 7, 6, 8};

    // output input array
    cout << "Array: ";
    for (int i = 0; i < N; i++) {
       cout << list[i] << " "; 
    } cout << endl;

    cout << "Bogo Sort" << endl;
    clock_t start = clock();
    bogo_sort(list);
    double elapsed = (double) (clock() - start) / CLOCKS_PER_SEC;

    cout << "Time taken: " << elapsed << " seconds" << endl;

    return 0;
}

bool is_sorted(int *arr) {
    for (int i = 0; i < (N - 1); i++) {
        if (arr[i] > arr[i+1]) {
            return false;
        }
    } return true;
}

void bogo_sort(int *arr) {
    while (!is_sorted(arr)) {
        for (int i = N - 1; i > 0; i--) {
            int index = rand() % (i + 1);

            int temp = arr[index];
            arr[index] = arr[i];
            arr[i] = temp;
        }
    }
}

File: src/java/Driver.java
import java.util.Arrays;

public class Driver {
    static long startTime;
    static long endTime;
    static long elapsedTime;
    static double elapsedTimeInSeconds;

    public static void main(String[] args) {
        int[] list = {1,4,2,3,5,7,6,8};
        System.out.println("Array: " + Arrays.toString(list) +"\n");

        int[] temp1 = Arrays.copyOf(list, list.length);
        startTime = System.nanoTime();
        FlipSort.flipSort(temp1);
        endTime = System.nanoTime();
        elapsedTime = endTime - startTime;
        elapsedTimeInSeconds = elapsedTime / 1_000_000_000.0;
        System.out.println("Time taken: " + elapsedTimeInSeconds + " seconds \n");

        int[] temp0 = Arrays.copyOf(list, list.length);
        startTime = System.nanoTime();
        BogoSort.bogoSort(temp0);
        endTime = System.nanoTime();
        elapsedTime = endTime - startTime;
        elapsedTimeInSeconds = elapsedTime / 1_000_000_000.0;
        System.out.println("Time taken: " + elapsedTimeInSeconds + " seconds \n");

        int[] temp2 = Arrays.copyOf(list, list.length);
        startTime = System.nanoTime();
        temp2 = BogobogoSort.bogoBogoSort(temp2);
        endTime = System.nanoTime();
        elapsedTime = endTime - startTime;
        elapsedTimeInSeconds = elapsedTime / 1_000_000_000.0;
        System.out.println("BogoBogo Sort: " + Arrays.toString(temp2));
        System.out.println("Time taken: " + elapsedTimeInSeconds + " seconds\n");

        int[] temp3 = Arrays.copyOf(list, list.length);
        startTime = System.nanoTime();
        RoBogoSort.roBogoSort(temp2);
        endTime = System.nanoTime();
        elapsedTime = endTime - startTime;
        elapsedTimeInSeconds = elapsedTime / 1_000_000_000.0;
        System.out.println("BogoBogo Sort: " + Arrays.toString(temp2));
        System.out.println("Time taken: " + elapsedTimeInSeconds + " seconds\n");

        startTime = System.nanoTime();
        StalinSort.stalinSort(list);
        endTime = System.nanoTime();
        elapsedTime = endTime - startTime;
        elapsedTimeInSeconds = elapsedTime / 1_000_000_000.0;
        System.out.println("Time taken: " + elapsedTimeInSeconds + " seconds \n");

        startTime = System.nanoTime();
        BlackbeardSort.blackBeardSort(list);
        endTime = System.nanoTime();
        elapsedTime = endTime - startTime;
        elapsedTimeInSeconds = elapsedTime / 1_000_000_000.0;
        System.out.println("Time taken: " + elapsedTimeInSeconds + " seconds \n");

        startTime = System.nanoTime();
        IntelligentDesignSort.intelligentDesignSort(list);
        endTime = System.nanoTime();
        elapsedTime = endTime - startTime;
        elapsedTimeInSeconds = elapsedTime / 1_000_000_000.0;
        System.out.println("Time taken: " + elapsedTimeInSeconds + " seconds \n");
    }
}
